examples/is31fl3741_native.py

```diff
@@ -0,0 +1,76 @@
+# SPDX-FileCopyrightText: 2021 Mark Komus
+# SPDX-License-Identifier: MIT
+import is31fl3741
+import displayio
+import framebufferio
+import board
+import busio
+from adafruit_display_text import label
+from adafruit_bitmap_font import bitmap_font
+from adafruit_led_animation.animation.comet import Comet
+from adafruit_led_animation.animation.chase import Chase
+from adafruit_is31fl3741.is31fl3741_PixelBuf import IS31FL3741_PixelBuf
+from adafruit_is31fl3741.led_glasses_map import (
+    glassesmatrix_ledmap_no_ring,
+    left_ring_map_no_inner,
+    right_ring_map_no_inner,
+)
+
+# Release any existing displays
+displayio.release_displays()
+
+# Create our own I2C bus with a 1Mhz frequency for faster updates
+i2c = busio.I2C(board.SCL, board.SDA, frequency=1000000)
+
+# Initialize the IS31FL3741 displayio display
+is31 = is31fl3741.IS31FL3741(
+    width=54,
+    height=15,
+    i2c=i2c,
+    scale=True,
+    gamma=True,
+    mapping=glassesmatrix_ledmap_no_ring,
+)
+display = framebufferio.FramebufferDisplay(is31, auto_refresh=True)
+
+# Turn the brightness down
+is31.brightness = 0.1
+
+# Create pixel buffers for each eye. Init is False as the display setup initialized the chip
+eye_left = IS31FL3741_PixelBuf(
+    i2c, left_ring_map_no_inner, 24, init=False, auto_write=False
+)
+eye_right = IS31FL3741_PixelBuf(
+    i2c, right_ring_map_no_inner, 24, init=False, auto_write=False
+)
```

Comment:
I think it'd be better if PixelBuf took in the IS31FL3741 object. That'll make it clearer when its access is shared.

```suggestion
# Initialize the IS31FL3741 displayio display
is31 = is31fl3741.IS31FL3741(
    width=54,
    height=15,
    i2c=i2c,
    scale=True,
    gamma=True,
    mapping=glassesmatrix_ledmap_no_ring,
)
display = framebufferio.FramebufferDisplay(is31, auto_refresh=True)

# Turn the brightness down
is31.brightness = 0.1

# Create pixel buffers for each eye. Init is False as the display setup initialized the chip
eye_left = IS31FL3741_PixelBuf(
    is31, left_ring_map_no_inner, 24, init=False, auto_write=False
)
eye_right = IS31FL3741_PixelBuf(
    is31, right_ring_map_no_inner, 24, init=False, auto_write=False
)
```

Do you need a separate `n` argument or could it be the length of the mapping?

Reply:
> I think it'd be better if PixelBuf took in the IS31FL3741 object. That'll make it clearer when its access is shared.

I spent some time thinking about this. The _PixelBuf object could be used when you won't otherwise have the is31 object (e.g. maybe a new board that is just a strip of LEDs you dont need a framebuffer for but want LED animations on). I do like the idea passing the is31 object. Maybe we want both options? For solo use or when it is combined? Any thoughts on that?
 
> Do you need a separate `n` argument or could it be the length of the mapping?

Good idea, I just had it because the pixelbuf object did but you are right I can derive it from the mapping.

Reply:
Thinking about this a bit more I think there is some confusion caused by this:

`IS31FL3741_PixelBuf` is just an implementation of the PixelBuf native class, using a native `is31fl3741_write` same as is used with NeoPixels. This works well for strings of LEDs driven by the IS31FL3741 like the eye lights

`is31fl3741.IS31FL3741` is the native framebuffer implementation that allows a matrix display to work on displayio (such as the main area of the LED glasses)

Both classes require the I2C bus to access the driver chip. What the example is doing is using both together at once, so yes the chip is shared. Maybe I should create an `LED_Glasses_Native` class that encapsulates both these objects into one, similar to the existing `LED_Glasses` class. Then the sharing of the I2C bus is hidden in that new class.

Reply:
Maybe what is needed is a native IS31FL3741 class that doesn't have a framebuffer. It could be shared between the framebuffer and pixelbuf classes. Framebuffer could be a second class in the `is31fl3741` module if it needs to be native.

Reply:
So just to make sure I'm clear in my own head a single class that "is" the IS31FL3741 chip. And then other classes (PixelBuf and FrameBuffer) will get passed the IS31FL3741 class.

Something like this?

```python
is31 = is31fl3741.IS31FL3741_Driver(i2c, address)

is31_pixelbuf = is31fl3741.IS31FL3741_PixelBuf(is31, ...)

is31_framebuffer = is31fl3741.IS31FL3741_FrameBuffer(is31, ...)
```

And determine the mix of native/python as required (off hand the driver would be native, pixelbuf would be in python and framebuffer TBD.

Reply:
Yup! That's what I'm thinking. I'd probably omit the `_Driver` suffix and then omit the `IS31FL3741_` prefixes. The module name already has it.